bottomless/src/replicator.rs

```diff
@@ -1402,38 +1398,33 @@ impl Replicator {
                     }
                 }
                 let frame = self.get_object(key.into()).send().await?;
-                let mut frameno = first_frame_no;
                 let mut reader = BatchReader::new(
-                    frameno,
+                    first_frame_no,
                     tokio_util::io::StreamReader::new(frame.body),
                     self.page_size,
                     compression_kind,
                 );
 
                 while let Some(frame) = reader.next_frame_header().await? {
-                    let pgno = frame.pgno();
-                    let page_size = self.page_size;
+                    last_received_frame_no = reader.next_frame_no();
                     reader.next_page(&mut page_buf).await?;
                     if self.verify_crc {
                         checksum = frame.verify(checksum, &page_buf)?;
                     }
-                    pending_pages.insert(pgno, &page_buf).await?;
-                    if frame.is_committed() {
-                        let pending_pages = std::mem::replace(
-                            &mut pending_pages,
-                            TransactionPageCache::new(
```

Comment:
If we're not using `TransactionPageCache` here I think it can be fully removed.

Reply:
we can't, it's still temporarily used in `bottomless-cli`. It will go away, because it should be rewritten to support encryption, but isn't yet